openpype/hosts/substancepainter/api/pipeline.py

```diff
@@ -0,0 +1,328 @@
+# -*- coding: utf-8 -*-
+"""Pipeline tools for OpenPype Gaffer integration."""
+import os
+import sys
+import logging
+from functools import partial
+
+# Substance 3D Painter modules
+import substance_painter.ui
+import substance_painter.event
+import substance_painter.project
+
+from openpype.host import HostBase, IWorkfileHost, ILoadHost, IPublishHost
+
+import pyblish.api
+
+from openpype.pipeline import (
+    register_creator_plugin_path,
+    register_loader_plugin_path,
+    AVALON_CONTAINER_ID
+)
+from openpype.lib import (
+    register_event_callback,
+    emit_event,
+)
+from openpype.pipeline.load import any_outdated_containers
+from openpype.hosts.substancepainter import SUBSTANCE_HOST_DIR
+
+log = logging.getLogger("openpype.hosts.substance")
+
+PLUGINS_DIR = os.path.join(SUBSTANCE_HOST_DIR, "plugins")
+PUBLISH_PATH = os.path.join(PLUGINS_DIR, "publish")
+LOAD_PATH = os.path.join(PLUGINS_DIR, "load")
+CREATE_PATH = os.path.join(PLUGINS_DIR, "create")
+INVENTORY_PATH = os.path.join(PLUGINS_DIR, "inventory")
+
+
+OPENPYPE_METADATA_KEY = "OpenPype"
+OPENPYPE_METADATA_CONTAINERS_KEY = "containers"  # child key
+
+self = sys.modules[__name__]
+self.menu = None
+self.callbacks = []
```

Comment:
Why not store them to object of `SubstanceHost`?

Reply:
Should that store it to the class of `SubstanceHost` or can it store it on the instance of `SubstanceHost`? :) I don't the instance itself stays around - or does the actual Host instance itself get stored as the `registered_host`?